src/storage/cloudStorage.ts

```diff
@@ -63,18 +72,28 @@ export const sortAccounts = (
 ) => {
   const acctList = values(accts)
   const sortedByAlias = sortBy(acctList, 'alias') || []
+  // if account is keystone account, add solanaAddress
+  const sortedByAliasWithSolanaAddress = sortedByAlias.map((acct) => {
+    // here we need to check if the account is keystone account
+    if (acct.keystoneDevice) {
+      return { ...acct, solanaAddress: acct.address }
```

Comment:
Why does this need to happen?

Reply:
I encountered a very strange problem. When using the Keystone cold wallet to import a Solana account, the data was stored completely, meaning the solanaAddress had a value. However, when the onboarding process ended and I tried to read the data on the account interface, I found that the solanaAddress had somehow become an empty string ("") for an unknown reason. To solve this strange issue, I intercepted the data before it was loaded from storage into the provider, and manually assigned the address value. This way, it could be displayed normally.

Reply:
What do you think might be the reason for this phenomenon? I'm looking forward to your response.

Reply:
It may be that you aren't properly writing a helium version of the solana address in the account? So some code later that write solanaAddress as a derivation of the `address` field writes empty?

Reply:
i am a bit confused. Could you tell me where the logic for changing the solanaAddress to an empty string is? Can you provide a link to it? I haven't been able to find it. I'd really appreciate it if you could.

Reply:
See the above comment on not setting the proper helium `address`

Reply:
I wrote a demo to convert a Solana wallet public key to a Helium old address. Is this conversion logic correct?

```
// convet solana public key to helium wallet address and back
const { PublicKey } = require("@solana/web3.js");
const { ED25519_KEY_TYPE } = require("@helium/address/build/KeyTypes");
const  Address = require("@helium/address").default;

// solana wallet path
// "m/44'/501'/0'"
// solana wallet public key
// "2b8d8b3addd92759f55b840c2852f5bd50aee3552fe987ee0d4fe24b9043df8e"
// solana wallet address
// 3w1iMvjKGxpbGaaSekNUsZBcVKERg2BCsUZMGrjcTMsj
let solWalletPublicKey = Buffer.from("2b8d8b3addd92759f55b840c2852f5bd50aee3552fe987ee0d4fe24b9043df8e", "hex");
let solWalletAddress = new PublicKey(solWalletPublicKey).toBase58();
console.log(solWalletAddress) // 3w1iMvjKGxpbGaaSekNUsZBcVKERg2BCsUZMGrjcTMsj


// convert solana public key to helium wallet address
const pkey = new PublicKey(solWalletPublicKey);
const heliumAddr = new Address(0, 0, ED25519_KEY_TYPE, pkey.toBytes());
let heliumAddress = heliumAddr.b58;
console.log(heliumAddress); // 13GvgGD1zwkBw5ZihDvXz81iMiJG97o3xL5Sz1wW2ffUUNHySxX


// convert helium wallet address to solana public key
const addr = Address.fromB58(heliumAddress);
let solanaAddress = new PublicKey(addr.publicKey).toBase58();
console.log(solanaAddress); // 3w1iMvjKGxpbGaaSekNUsZBcVKERg2BCsUZMGrjcTMsj
```

If this logic is correct, then I will modify the logic in the above code. In other words, when I get the Solana public key from the keystone cold wallet, I need to first convert it to a Helium old address and assign it to the 'address' variable. The 'solanaAddress' will be automatically assigned after completing the migrate operation.

Reply:
So, about what you said regarding some code that resets the solanaAddress to empty - I've been trying to find this piece of logic. Can you tell me where it is? I've been looking all over for it but can't seem to locate it.